### PR TITLE
3.0: Check block device has Ebs section to get SnapshotId

### PR DESCRIPTION
**Pull request by yuleiwan**
Block device mapping returns device info
`[ {'DeviceName': '/dev/sda1', 'Ebs': {'DeleteOnTermination': True, 'SnapshotId': 'snap-042b224dde6bd4030', 'VolumeSize': 50, 'VolumeType': 'gp2', 'Encrypted': False}}, 
{'DeviceName': '/dev/sdb', 'VirtualName': 'ephemeral0'}, 
{'DeviceName': '/dev/sdc', 'VirtualName': 'ephemeral1'} ]`
We should check Ebs section is in the array entity, otherwise, NoneType error will raise like 'NoneType' object has no attribute 'get'

Signed-off-by: Yulei Wang <yuleiwan@amazon.com>

**Please See** [Git Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions)

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by hanwen-pcluste — APPROVED** [review]
LGTM!

File: cli/src/pcluster/aws/aws_resources.py [hunk 1]
```diff
@@ -309,7 +309,8 @@ def snapshot_ids(self) -> list:
         """Return snapshot ids."""
         snapshot_ids = []
         for block_device_mapping in self.block_device_mappings:
-            snapshot_ids.append(block_device_mapping.get("Ebs").get("SnapshotId"))
+            if block_device_mapping.get("Ebs"):
```
Comment (enrico-usai) [inline cli/src/pcluster/aws/aws_resources.py:312]: I think you should use: `if block_device_mapping.get("Ebs", None):`to cover the case on which Ebs is not defined.

Comment (hanwen-pcluste) [reply cli/src/pcluster/aws/aws_resources.py:312]: Isn't `if block_device_mapping.get("Ebs", None)` equal to `if block_device_mapping.get("Ebs")`

Comment (enrico-usai) [reply cli/src/pcluster/aws/aws_resources.py:312]: You're right! None is the default returned value.

**Pull request merged**